src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -958,16 +958,25 @@ public ProjectState UpdateDocumentsOrder(ImmutableList<DocumentId> documentIds)
                     _lazyLatestDocumentTopLevelChangeVersion;
         }
 
-        private sealed class DocumentIdComparer : IComparer<DocumentId>
+        private sealed class DocumentIdComparer : IComparer<DocumentId?>
         {
-            public static IComparer<DocumentId> Instance = new DocumentIdComparer();
+            public static readonly IComparer<DocumentId?> Instance = new DocumentIdComparer();
 
             private DocumentIdComparer()
             {
             }
 
-            public int Compare(DocumentId x, DocumentId y)
+            public int Compare(DocumentId? x, DocumentId? y)
             {
+                if (x is null)
```

Comment:
Given the code that uses this, if we get a null that means something went Really Terribly Wrong™. Can we just throw on null or annotate this particular one differently?

Reply:
No. The framework _requires_ that implementations accept null.

Reply:
https://github.com/dotnet/runtime/blob/9981b1f128283c9fcff32689f34bf51289209b03/src/libraries/System.Private.CoreLib/src/System/IComparable.cs#L37